MathLib/LinAlg/RowColumnIndices.h

```diff
@@ -14,19 +14,25 @@
 
 namespace MathLib
 {
+// A dummy vector as the default value of the third argument of
+// the constructor.
+static std::vector<std::size_t> dummy_non_ghost_local_ids;
```

Comment:
All reference member can only be initialized in the initialization list. Therefore, overload ctor with different argument is also not possible. Alternative, an ifdef as the previous commit or some other solution that I am thinking about.